src/lib/Form/Type/Invitation/RoleChoiceType.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\User\Form\Type\Invitation;
+
+use Ibexa\Contracts\Core\Repository\PermissionResolver;
+use Ibexa\Contracts\Core\Repository\RoleService;
+use Ibexa\Core\Repository\Repository;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\ChoiceList\ChoiceList;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class RoleChoiceType extends AbstractType
+{
+    private RoleService $roleService;
+
+    private Repository $repository;
+
+    private PermissionResolver $permissionResolver;
+
+    public function __construct(
+        RoleService $roleService,
+        Repository $repository,
+        PermissionResolver $permissionResolver
+    ) {
+        $this->roleService = $roleService;
+        $this->repository = $repository;
+        $this->permissionResolver = $permissionResolver;
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefaults([
+                'choice_loader' => ChoiceList::lazy(
+                    $this,
+                    fn () => $this->loadFilteredRoles(),
```

Comment:
Missing third argument (`$vary`). It''s not immediately visible here, but this choice list depends on current user context.

This means that this form type, used elsewhere on page, may contain incorrect list.

Reply:
All changed to `CallbackChoiceLoader`.